src/app/ui-components/sidebar-right/sidebar-right.component.html

```diff
@@ -1,32 +1,123 @@
+                                attr.aria-labelledby="dropdownbutton-{{sidebarElement.getName()}}">
+                                <button
+                                    id="edit"
+                                    ngbDropdownItem
+                                    (click)="calbackMethods[1].clickCallback(sidebarElement.getUUID())">
+                                    <img
+                                        class="item pr-1 mb-1"
+                                        *ngIf="calbackMethods[1].icon !== ''"
+                                        [src]="calbackMethods[1].icon" />
+                                    <label>{{calbackMethods[1].label}}</label>
+                                </button>
+                                <div *ngIf="lStorage == 'chat'">
+                                    <button
+                                        id="export"
+                                        ngbDropdownItem
+                                        (click)="calbackMethods[3].clickCallback(sidebarElement.getUUID())">
+                                        <img
+                                            class="item pr-1 mb-2"
+                                            *ngIf="calbackMethods[3].icon !== ''"
+                                            [src]="calbackMethods[3].icon" />
+                                        <label
+                                            >{{calbackMethods[3].label}}</label
+                                        >
+                                    </button>
+                                </div>
+                                <button
+                                    id="delete"
+                                    ngbDropdownItem
+                                    (click)="calbackMethods[2].clickCallback(sidebarElement.getUUID())">
+                                    <img
+                                        class="item pr-1 mb-1"
+                                        *ngIf="calbackMethods[2].icon !== ''"
+                                        [src]="calbackMethods[2].icon" />
+                                    <label>{{calbackMethods[2].label}}</label>
+                                </button>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div class="row mt-4">
+                <div class="col-11 ml-3 element-div">
+                    <div class="row">
+                        <button
+                            type="button"
+                            class="col btn pl-3 p-3 text-truncate element text-center"
+                            id="add-new-button"
+                            (click)="calbackMethods[0].clickCallback('')">
+                            <label class="mb-0"
+                                >{{calbackMethods[0].label}}</label
+                            >
+                        </button>
+                    </div>
```

Comment:
maybe create a separate `@Input` field in the sidebar-right-component where you can specify the 'add-callback' (i.e. what is currently calbackMethods[0]), to make it more clear, that this one is handled differently from the other methods in the ui. The 'calbackMethods' array could then maybe be renamed to 'dropdownCallbackMethods' or something similar.

Reply:
resolved